export_params/client.sh

```diff
@@ -0,0 +1,223 @@
+#!/bin/sh
+#
+# Environment variables used:
+#  - SERVER: hostname or IP-address of the NFS-server
+#  - EXPORT: NFS-export to test (should start with "/")
+
+# enable some more output
+set -x
+
+[ -n "${SERVER}" ]
+[ -n "${EXPORT}" ]
+
+if [ "$1" = "client_initialization" ]
+then
+	# install build and runtime dependencies
+	yum -y install nfs-utils time
+
+	mkdir -p /mnt/ganesha
+
+	mount -t nfs -o vers=3 ${SERVER}:${EXPORT} /mnt/ganesha
+
+	echo "Client Initial Stage --- With All Rights To All Clients ( RO & RW ) "
+
+	cd /mnt/ganesha
+	
+	echo "Trying To Write A File"
+	echo "Hello World" > testFile.txt
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "SUCCESS ON WRITING RIGHTS"
+	else
+		echo "FAILED ON WRITING RIGHTS"
+		exit ret
+	fi
+
+	echo "Trying To Read A File"
+	cat testFile.txt
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "SUCCESS"
+	else
+		echo "FAILED ON READING RIGHTS"
+		exit ret
+	fi
+
+	echo "Trying To Change File Ownership For Checking ROOT Rights"
+	sudo chown root testFile.txt
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "SUCCESS"
+	else
+		echo "FAILED ON ROOT RIGHTS"
+		exit ret
+	fi
+
+	cd / && umount /mnt/ganesha
+fi
+
+if [ "$1" = "client_stage1" ]
+then
+	echo "Client Stage 1 --- With Only RO Rights To Clients "
+
+	mount -t nfs ${SERVER}:${EXPORT} /mnt/ganesha
+
+	cd /mnt/ganesha
+
+	echo "Trying To Write A File"
+	sed -i '1s/$/ From RedHat/' testFile.txt
+	ret=$?
+	if [ $ret -eq 0 ]
+	then
+		echo "FAILURE Since Write Permissions Were Not Blocked To The Client"
+		exit ret
+	else
+		echo "SUCCESS ON WRITE PERMISSIONS FAILURE"
```

Comment:
This should probably not include the word "FAILURE"

Reply:
@dang 
What I meant here is ... The expected output is that the write permissions should fail .... 
Should I give a better wording like ... "SUCCESS ON WRITE PERMS DECLINED" ?

Reply:
That would be fine, yes.  It's just that people (and CI frameworks) frequently look for FAIL and FAILURE in output, so see if a test failed.  In this case, it is succeeding, so it shouldn't use the word FAIL.  Anything else (like DECLINED) is fine.